sensors/motionsensor.cpp

```diff
@@ -0,0 +1,26 @@
+#include "motionsensor.h"
+#include <iostream>
+
+namespace smartHome {
+    MotionSensor::MotionSensor(const std::string& sensorId, double value, bool isFunctional)
+        : Sensor("Motion", sensorId, value, isFunctional) {}
+
+    void MotionSensor::measure() {
+
+        if(!getIsFunctional()){
+            std::cout << "The sensor is not working properly!!" << std::endl;
+            return;
+        }
+
+        std::cout << "Detecting motion..." << std::endl;
+        
+        // Assume a simple simulation where the sensor value is toggled between 0 and 1
+        if (getValue() == 0.0) {
```

Comment:
It's not a good practice to compare floating point numbers (float and double) as we compare integers. In your case chances are you won't have issues but in real world application this is a bug waiting to happen. If you have time after finishing the other comments, read about comparing floating point numbers (not just C++, all languages should have the same problem).